ddsenabler/src/cpp/DDSEnabler.cpp

```diff
@@ -218,5 +292,363 @@ bool DDSEnabler::publish(
     return enabler_participant_->publish(topic_name, json);
 }
 
+bool DDSEnabler::send_service_request(
```

Comment:
I suggest moving all this RPC specific logic to be in the enabler participant, I don't see any need for having it separated.